vmm/src/device_manager.rs

```diff
@@ -2116,6 +2190,37 @@ impl DeviceManager {
             };
         }
 
+        let debugcon_config = self.config.lock().unwrap().debugcon.clone();
```

Comment:
This part unfortunately have lots of code duplication with the setup of the serial console and the virtio console. I tried a few things, but didn't find a way to simplify without a major refactoring.